lib/ReadsUtils/ReadsUtilsImpl.py

```
@@ -642,6 +680,330 @@ def get_fq_stats(self, reads_object, file_path):
+        """
+        _download_web_file: download reads source file from web
+
+        file_url: file URL
+        download_type: one of ['Direct Download', 'FTP', 'DropBox', 'Google Drive']
+        rev_file: optional, default as False. Set to True if file_url is for rev_file
+
+        return: file path of downloaded web file
+
+        """
+
+        # prepare local copy file path for copy
+        self.log('Start downloading web file from: %s' % file_url)
+        tmp_file_name = 'tmp_rev_fastq.fastq' if rev_file else 'tmp_fwd_fastq.fastq'
+        dstdir = os.path.join(self.scratch, 'tmp')
+        if not os.path.exists(dstdir):
+            os.makedirs(dstdir)
+        copy_file_path = os.path.join(dstdir, tmp_file_name)
+
+        self._download_file(download_type, file_url, copy_file_path)
+        self.log('Copied web file to %s' % copy_file_path)
+
+        return copy_file_path
+
+
+    def _download_file(self, download_type, file_url, copy_file_path):
+        """
+        _download_file: download execution distributor
+
+        params:
+        download_type: download type for web source file
+        file_url: file URL
+        copy_file_path: output file saving path
+        
+        """
+        valid_download_type = ['Direct Download', 'DropBox', 'FTP', 'Google Drive']
+        if download_type not in valid_download_type:
+            raise ValueError('Invalid download type: %s. Please use one of %s' % (download_type, valid_download_type))
+
+        if download_type == 'Direct Download':
+            self._download_direct_download_link(file_url, copy_file_path)
+        elif download_type == 'DropBox':
+            self._download_dropbox_link(file_url, copy_file_path)
+        elif download_type == 'FTP':
+            self._download_ftp_link(file_url, copy_file_path)
+        elif download_type == 'Google Drive':
+            self._download_google_drive_link(file_url, copy_file_path)
```

Still need to throw an error here

---

👍